examples/client-server/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/xdire/xlb"
+	testUtil "github.com/xdire/xlb/httputil"
+	"github.com/xdire/xlb/tlsutil"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+func main() {
+
+	ctx, cancelCtx := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
+	defer cancelCtx()
+
+	// Flags for pre-generate TLS data and identity
+	identity := flag.String("id", "test", "Identity name to match for certificates")
+	optionGenTLS := flag.Bool("gentls", false, "Generate TLS data for all parts of the chain to run")
+	optionWipeTLS := flag.Bool("wipetls", false, "Wipe TLS data in the folder")
+
+	// Flags for launching the load balancer service
+	optionLoadBalancer := flag.Bool("start-lb", false, "Launch the load balancer service")
+	port := flag.Int("port", 9089, "Port for the load balancer service")
+	rateQuota := flag.Int("rate-quota", 10, "Rate quota for the load balancer")
+	rateDuration := flag.Duration("rate-duration", time.Second, "Rate duration for the load balancer")
+	serverCert := flag.String("server-cert", "server.crt", "Server certificate file for the load balancer")
+	serverKey := flag.String("server-key", "server.key", "Server key file for the load balancer")
+	routesStr := flag.String("routes", "", "Routes for the load balancer in the format 'localhost:9081,localhost:9082,localhost:9083'")
+
+	// Flag for launching the SDK client
+	optionClient := flag.Bool("send-client", false, "Launch the client to reach servers behind the load balancer")
+	optionClientThreads := flag.Int("threads", 3, "How many parallel clients should be there trying to reach the servers behind the LB")
+
+	// Flag for launching test servers
+	optionTestServers := flag.Bool("start-test-servers", false, "Launch test servers to be hidden behind the load balancer, must be in conjunction with --routes xxx:port,xxx:port")
+
+	flag.Parse()
+
+	// Prepare routes from flags
+	routes := strings.Split(*routesStr, ",")
+	pRoutes := make([]xlb.Route, 0)
+	for _, rte := range routes {
+		if len(rte) == 0 {
+			continue
+		}
+		pRoutes = append(pRoutes, xlb.ServicePoolRoute{
+			ServicePath:   rte,
+			ServiceActive: true,
+		})
+	}
+
+	// Generate TLS data in folder
+	if *optionGenTLS {
+		err := tlsutil.CreateLocalTLSData(*identity)
+		if err != nil {
+			log.Fatalf("cannot generate TLS data, error: %+v", err)
+		}
+	}
+
+	// Clean TLS data in folder
+	if *optionWipeTLS {
+		removed, err := tlsutil.WipeLocalTLSData("./")
+		if err != nil {
+			log.Fatalf("cannot generate TLS data, error: %+v", err)
+		}
+		log.Printf("files deleted: %+v", removed)
+	}
+
+	// Launch load balancer using provided options
+	if *optionLoadBalancer {
+
+		if len(pRoutes) == 0 {
+			log.Fatalf("no routes provided")
+			return
+		}
+
+		cert, err := os.ReadFile(*serverCert)
+		if err != nil {
+			log.Fatalf("server certificate not found")
```

Comment:
I don't think it's necessarily safe to assume that this is a "not found" - this could just as easily be a permissions error.